### ISSUE TITLE
Find list and grouping broken

### ISSUE THREAD

**Issue by dereuromark** [cakephp/cakephp#12922]
This is a (multiple allowed):

* [x] bug
* [ ] enhancement
* [ ] feature-discussion (RFC)

* CakePHP Version: latest 3.7

### What you did
We had this query in our app:

```php
$array = $this->find()
            ->select(['Modules.id', 'Modules.name', 'Modules.type'])
            ->contain('Modules')
            ->order(['type' => 'asc', 'name' => 'asc'])
            ->distinct()
            ->find('list', ['groupField' => 'Modules.type', 'keyField' => 'Modules.id', 'valueField' => 'Modules.name'])->toArray();
```

Also distinct(['Modules.id', 'Modules.name', 'Modules.type']) - which produces a group by instead of distinct has the same issue.
Seems like this is a problem with the find(list) behavior of grouping then.

### What happened
It now just outputs
```
array:1 [▼
  "" => array:1 [▼
    "" => null
  ]
]
```

### What you expected to happen
Before it was a list of the modules grouped by their type for use in select forms in search filtering.

This must have been broken recently, as it used to work for sure in 3.6, maybe even afterwards.

**Comment by dereuromark** [comment]
Darn, these cases are also a side-effect of the breaking changes in 3.7...
We might have hundreds of those silent cases of BC breaks in all our apps here :/
```php
->find('list', ['groupField' => 'module.type', 'keyField' => 'module.id', 'valueField' => 'module.name'])->toArray();
```
actually works again. But since there is no notice/error these are hard to spot programmatically.

I wonder if those could be shimmed somehow - or get warnings here at least in some way inside e.g. integration testing.

**Comment by markstory** [comment]
This sounds very similar to #12880 and could be related to the contain bug that was fixed in 3.6.14.

**Comment by dereuromark** [comment]
It is the same issue, but very silent in its behavior change, so it often goes unnoticed.
I am ok to manually search and fix any of these upgrade issues. I was just very suprised to have broken forms everywhere without knowing it.
